test/src/math/test_vecmat.cpp

```diff
@@ -0,0 +1,487 @@
+
+			ASSERT_FLOAT_EQ(value3, value1 - value2);
+		}
+	}
+}
+
+TEST_F(VecmatTest, test_vm_vec_add2)
+{
+	for (size_t loop = 0; loop < 1000; ++loop) {
+		vec3d v1, v2, v1backup;
+
+		static_randvec_unnormalized(rand32(), &v1);
+		static_randvec_unnormalized(rand32(), &v2);
+
+		v1backup.xyz = v1.xyz;
+
+		vm_vec_add2(&v1, &v2);
+
+		for (size_t i = 0; i < 3; ++i) {
+			auto value1 = v1.a1d[i];
+			auto value2 = v2.a1d[i];
+			auto value3 = v1backup.a1d[i];
+
+			ASSERT_FLOAT_EQ(value1, value2 + value3);
+		}
+	}
+}
+
+TEST_F(VecmatTest, test_vm_vec_sub2)
+{
+	for (size_t loop = 0; loop < 1000; ++loop) {
+		vec3d v1, v2, v1backup;
+
+		static_randvec_unnormalized(rand32(), &v1);
+		static_randvec_unnormalized(rand32(), &v2);
+
+		v1backup.xyz = v1.xyz;
+
+		vm_vec_sub2(&v1, &v2);
+
+		for (size_t i = 0; i < 3; ++i) {
+			auto value1       = v1.a1d[i];
+			auto value2       = v2.a1d[i];
+			auto value1Backup = v1backup.a1d[i];
+
+			ASSERT_FLOAT_EQ(value1, value1Backup - value2);
+		}
+	}
+}
+
+TEST_F(VecmatTest, test_vm_vec_avg)
+{
+	for (size_t loop = 0; loop < 1000; ++loop) {
+		vec3d vArray[2];
+		vec3d vAvg;
+
+		for (size_t i = 0; i < 2; i++) {
+			static_randvec_unnormalized(rand32(), &vArray[i]);
+		}
+
+		vm_vec_avg(&vAvg, &vArray[0], &vArray[1]);
+
+		float accX = 0.0f, accY = 0.0f, accZ = 0.0f;
+
+		for (size_t i = 0; i < 2; ++i) {
+			accX += vArray[i].xyz.x;
+			accY += vArray[i].xyz.y;
+			accZ += vArray[i].xyz.z;
+		}
+
+		accX /= 2;
+		accY /= 2;
+		accZ /= 2;
+
+		ASSERT_FLOAT_EQ(vAvg.xyz.x, accX);
+		ASSERT_FLOAT_EQ(vAvg.xyz.y, accY);
+		ASSERT_FLOAT_EQ(vAvg.xyz.z, accZ);
+	}
+}
+
+TEST_F(VecmatTest, test_vm_vec_avg3)
+{
+	for (size_t loop = 0; loop < 1000; ++loop) {
+		vec3d vArray[3];
+		vec3d vAvg;
+
+		for (size_t i = 0; i < 3; i++) {
+			static_randvec_unnormalized(rand32(), &vArray[i]);
+		}
+
+		vm_vec_avg3(&vAvg, &vArray[0], &vArray[1], &vArray[2]);
+
+		float accX = 0.0f, accY = 0.0f, accZ = 0.0f;
+
+		for (size_t i = 0; i < 3; ++i) {
+			accX += vArray[i].xyz.x;
+			accY += vArray[i].xyz.y;
+			accZ += vArray[i].xyz.z;
+		}
+
+		accX /= 3;
+		accY /= 3;
+		accZ /= 3;
+
+		ASSERT_FLOAT_EQ(vAvg.xyz.x, accX);
+		ASSERT_FLOAT_EQ(vAvg.xyz.y, accY);
+		ASSERT_FLOAT_EQ(vAvg.xyz.z, accZ);
+	}
+}
+
+TEST_F(VecmatTest, test_vm_vec_avg4)
+{
+	for (size_t loop = 0; loop < 1000; ++loop) {
+		vec3d vArray[4];
+		vec3d vAvg;
+
+		for (size_t i = 0; i < 4; i++) {
+			static_randvec_unnormalized(rand32(), &vArray[i]);
+		}
+
+		vm_vec_avg4(&vAvg, &vArray[0], &vArray[1], &vArray[2], &vArray[3]);
+
+		float accX = 0.0f, accY = 0.0f, accZ = 0.0f;
+
+		for (size_t i = 0; i < 4; ++i) {
+			accX += vArray[i].xyz.x;
+			accY += vArray[i].xyz.y;
+			accZ += vArray[i].xyz.z;
+		}
+
+		accX /= 4;
+		accY /= 4;
+		accZ /= 4;
+
+		ASSERT_FLOAT_EQ(vAvg.xyz.x, accX);
+		ASSERT_FLOAT_EQ(vAvg.xyz.y, accY);
+		ASSERT_FLOAT_EQ(vAvg.xyz.z, accZ);
+	}
+}
+
+TEST_F(VecmatTest, test_vm_vec_avg_n)
+{
+	for (size_t loop = 0; loop < 1000; ++loop) {
```

Comment:
All the tests for the random functionality could be improved by using a value parameterized test since that would avoid having the `for` loop in all the test cases.